packages/@aws-cdk/aws-lambda-nodejs/test/integ.dependencies.ts

```diff
@@ -24,6 +27,40 @@ class TestStack extends Stack {
   }
 }
 
+class SdkV3TestStack extends Stack {
+  public lambdaFunction: IFunction
+
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    // This function uses @aws-sdk/* but it will not be included
+    this.lambdaFunction = new lambda.NodejsFunction(this, 'external-sdk-v3', {
+      entry: path.join(__dirname, 'integ-handlers/dependencies-sdk-v3.ts'),
+      runtime: Runtime.NODEJS_18_X,
+    });
+  }
+}
+
 const app = new App();
-new TestStack(app, 'cdk-integ-lambda-nodejs-dependencies');
+const sdkV2testCase = new SdkV2TestStack(app, 'cdk-integ-lambda-nodejs-dependencies');
+const sdkV3testCase = new SdkV3TestStack(app, 'cdk-integ-lambda-nodejs-dependencies-for-sdk-v3');
+
+for (const testCase of [sdkV2testCase, sdkV3testCase]) {
+  const integ = new IntegTest(app, `LambdaDependencies-${testCase.stackId}`, {
+    testCases: [testCase],
+  });
+
+  const response = integ.assertions.awsApiCall('Lambda', 'invoke', {
+    FunctionName: testCase.lambdaFunction.functionName,
+  });
+  response.provider.addToRolePolicy({
+    Effect: 'Allow',
+    Action: ['lambda:InvokeFunction'],
+    Resource: [testCase.lambdaFunction.functionArn],
+  });
+  response.expect(ExpectedResult.objectLike({
+    StatusCode: 200,
+  }));
```

Comment:
Just rechecked this and Lambda always returns 200 if the "call" was successful, even when the function errors.

You need to check that there is no `FunctionError` in the response.